app/models/katello/authorization/content_view_version.rb

```diff
@@ -10,9 +10,32 @@ def all_hosts_editable?(lifecycle_environment)
     end
 
     module ClassMethods
+      def with_content_view_scope(scope)
+        joins(:content_view).where("#{Katello::ContentView.table_name}.id" => ::Katello::ContentView.send(scope))
```

Comment:
If you include the Authorizable module in here you can do this: `joins_authorized(::Katello::ContentView, :view_content_views)` for example

Reply:
@jlsherrill maybe you'd prefer the merge approach? `joins(:content_view).merge(Katello::ContentView.readable)` for example

Reply:
i'm hesitant to add Authorizable due to the check_permissions_after_save hook it adds, but i like .merge()